app.js

```diff
@@ -180,9 +180,58 @@ class FilterBox1 {
   }
 }
 
-const filterBox1 = new FilterBox1();
+const filterBox1 = new FilterBoxOpenAndClose();
 
-class FilterBox2 {}
+class FilterByType {
+  constructor() {
+    this.onlineCheckbox = document.querySelector("#onlineChallengeCheckbox");
+    this.onsiteCheckbox = document.querySelector("#onsiteChallengeCheckbox");
+
+    this.onlineCheckbox.addEventListener("change", () => {
+      this.filterChallengesByType();
+    });
+
+    this.onsiteCheckbox.addEventListener("change", () => {
+      this.filterChallengesByType();
+    });
+  }
+
+  filterChallengesByType() {
+    const onlineChecked = this.onlineCheckbox.checked;
+    const onsiteChecked = this.onsiteCheckbox.checked;
+
+    const challengesContainer = document.querySelector("#ourChallenges");
+    const api = new APIadapter();
+
+    api.fetchAllChallenges().then((challenges) => {
+      let filteredChallenges;
+
+      if (!onlineChecked && !onsiteChecked) {
+        // No checkboxes are checked, show all challenges
+        filteredChallenges = challenges;
+      } else {
+        // Filter challenges
+        filteredChallenges = challenges.filter((challenge) => {
+          return (
+            (onlineChecked && challenge.data.type === "online") ||
+            (onsiteChecked && challenge.data.type === "onsite")
+          );
+        });
+      }
+
+      // Clear challenges
+      challengesContainer.innerHTML = "";
+
+      // Render challenges
+      filteredChallenges.forEach((challenge) => {
+        const element = challenge.render();
+        challengesContainer.appendChild(element);
+      });
```

Comment:
This code resets the UI and re-renders challenges every time this filter changes. But it only considers _this filter_ (which makes sense, because the code is here in this filter class). That means that whatever filter you interact with _last_ will be the only filter that actually affects what's visible.

Thinking about separation of concerns, the fundamental problem here is that this class is supposed to be concerned with a single filter, but it also concerns itself with the full UI output. That's going to make it difficult to collaborate with other filters.